src/openzeppelin/tests/test_erc721.cairo

```diff
@@ -0,0 +1,482 @@
+use openzeppelin::introspection::erc165;
+use openzeppelin::token::erc721;
+use openzeppelin::token::erc721::ERC721;
+
+use starknet::contract_address_const;
+use starknet::ContractAddress;
+use starknet::testing::set_caller_address;
+use integer::u256;
+use integer::u256_from_felt252;
+use traits::Into;
+use zeroable::Zeroable;
+
+const NAME: felt252 = 111;
+const SYMBOL: felt252 = 222;
+const URI: felt252 = 333;
+
+fn TOKEN_ID() -> u256 {
+    7.into()
+}
+
+fn ZERO() -> ContractAddress {
+    Zeroable::zero()
+}
+fn OWNER() -> ContractAddress {
+    contract_address_const::<1>()
+}
+fn RECIPIENT() -> ContractAddress {
+    contract_address_const::<2>()
+}
+fn SPENDER() -> ContractAddress {
+    contract_address_const::<3>()
+}
+fn OPERATOR() -> ContractAddress {
+    contract_address_const::<4>()
+}
+fn OTHER() -> ContractAddress {
+    contract_address_const::<5>()
+}
+
+///
+/// Setup
+///
+
+fn setup() {
+    ERC721::initializer(NAME, SYMBOL);
+    ERC721::_mint(OWNER(), TOKEN_ID());
+}
+
+///
+/// Initialize
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_initialize() {
+    ERC721::initializer(NAME, SYMBOL);
+
+    assert(ERC721::name() == NAME, 'Name should be NAME');
+    assert(ERC721::symbol() == SYMBOL, 'Symbol should be SYMBOL');
+    assert(ERC721::balance_of(OWNER()) == 0.into(), 'Balance should be zero');
+
+    assert(ERC721::supports_interface(erc721::interface::IERC721_ID), 'Missing interface ID');
+    assert(
+        ERC721::supports_interface(erc721::interface::IERC721_METADATA_ID), 'missing interface ID'
+    );
+    assert(ERC721::supports_interface(erc165::IERC165_ID), 'missing interface ID');
+    assert(!ERC721::supports_interface(erc165::INVALID_ID), 'invalid interface ID');
+}
+
+///
+/// Getters
+///
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ERC721: invalid account', ))]
+fn test_balance_of_zero() {
+    ERC721::balance_of(ZERO());
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ERC721: invalid token ID', ))]
+fn test_owner_of_non_minted() {
+    ERC721::owner_of(u256_from_felt252(7));
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('ERC721: invalid token ID', ))]
+fn test_get_approved_nonexistent() {
+    ERC721::get_approved(u256_from_felt252(7));
+}
```

Comment:
i only added the panicking `token_uri` test since the non-panicking one was returning the weird withdraw_gas error, and it's already tested in `_set_token_id` anyway